modules/post/multi/manage/local_mitm.rb

```diff
@@ -0,0 +1,58 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Post
+  include Msf::Post::File
+  include Msf::Post::Windows::Priv
+
+  def initialize(info={})
+    super(update_info(info,
+      'Name'          => 'Local MITM',
+      'Description'   => %q{
+        This module setup a local proxy on the victim computer so you can MITM it.
+      },
+      'License'       => MSF_LICENSE,
+      'Author'        => [ 'Eliott Teissonniere' ],
+      'Platform'      => [ 'win', 'linux' ],
+      'SessionTypes'  => [ 'meterpreter', 'shell' ]
+    ))
+
+    register_options(
+      [
+        OptString.new('RHOST', [true, 'Host of the socks proxy server']),
+        OptString.new('RPORT', [true, 'Port of the socks proxy server']),
+        OptBool.new('CLEANUP', [false, 'If we should remove the proxy instead of installing it'])
```

Comment:
This might be better implemented as a module `ACTION`, with `START` / `STOP` options.

That would make the following module logic cleaner too.

Reply:
Ok will do!

Reply:
Cool. You can set a default action with `DefaultAction` in the module properties hash.